crates/contracts/src/event/event.cairo

```diff
@@ -0,0 +1,48 @@
+use dojo::meta::Layout;
+use dojo::meta::introspect::Ty;
+use dojo::world::IWorldDispatcher;
+
+pub trait Event<T> {
+    fn emit(self: @T, world: IWorldDispatcher);
+
+    fn name() -> ByteArray;
+    fn namespace() -> ByteArray;
+    fn tag() -> ByteArray;
+
+    fn version() -> u8;
+
+    fn selector() -> felt252;
+    fn instance_selector(self: @T) -> felt252;
+
+    fn name_hash() -> felt252;
+    fn namespace_hash() -> felt252;
+
+    fn layout() -> Layout;
+    fn schema(self: @T) -> Ty;
+
+    fn packed_size() -> Option<usize>;
+    fn unpacked_size() -> Option<usize>;
```

Comment:
This shouldn't make sense for an event, since we're never hitting the contract storage.

Reply:
Yes, you're right ! Thanks 🙏